mlog/logr.go

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mlog
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/wiggin77/logr"
+	logrFmt "github.com/wiggin77/logr/format"
+	"github.com/wiggin77/logr/target"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	DefaultMaxTargetQueue = 1000
+	DefaultSysLogPort     = 514
+)
+
+type LogLevel struct {
+	ID         logr.LevelID
+	Name       string
+	Stacktrace bool
+}
+
+type LogTarget struct {
+	Type         string // one of "console", "file", "tcp", "syslog".
+	Format       string // one of "json", "plain"
+	Levels       []LogLevel
+	Options      map[string]interface{}
```

Comment:
Have you thought about using https://golang.org/pkg/encoding/json/#RawMessage here instead? Then you could have proper options structs for each of the types.

Reply:
I had not considered RawMessage.  That is a great idea.  Thanks.

Reply:
Done via https://github.com/mattermost/mattermost-server/pull/14888/commits/ed75257ac07096ad28b2fc53101f3daf27bf01ac